### DISCUSSION TITLE
introductory question

### DISCUSSION THREAD

**Discussion by mcacace** [thelfer/MFrontGenericInterfaceSupport#71] [General]
Dear all, 
I started (two days ago) to interface mfront, via the generic support interface, to the multiphysics numerical framework MOOSE (here is the link to the web site - https://mooseframework.inl.gov/). This said, all works fine for the simple linear solid elastic model I did test - running time is also comparable to the native implementation. 
I was wondering however whether there is a more detailed documentation available about both the mfront library and the interface. I am not 100% sure if the way I am passing and retrieving the information to and from the two parties is the most efficient one - would be lovely to understand better the link between the gradients function of the interface and the mfront objects used in the behaviour description. 
I duplicated the same message to the contact email since I do not know which platform is the right one to use, sorry about that. 
Thanks a lot for any guidance,
Mauro

**Comment by thelfer** [comment]
Hi @mcacace,

`MOOSE` is a very interesting project, it would be very interesting to be able to couple `MOOSE` and `MFront` !

The most uptodate document to explore MGIS API is this document: 
<https://thelfer.github.io/mgis/web/bindings-cxx.html>

You can download as a PDF here: 
<https://www.researchgate.net/publication/333089236_A_brief_introduction_to_the_MGIS_C_library_for_mechanical_behaviours>

To be more precise, I would need more information about how `MOOSE` works, i.e.:

- how values at integrations points are handled
- how integration errors are handled
- how information can be retrieved
-  etc...

Do not hesitate to contact me in private if you want to organize a small meeting.

Best,
Thomas

**Reply by mcacace** [reply]
@thelfer, thanks for the quick reply and for the links! I will go through them. About your questions, I can provide some short answers (see below), but I think it might be best if we can schedule a small meeting. I might need to wait for a colleague of mine (Francesco Parisio, with whom you have collaborated in the past) to be back next week. 
About your points: 
What I did so far is to have mfront interfaced at the element quadrature points directly, this is native Moose to handle material properties (stress, strain etc..). Moose computes material on the fly, whenever you need them, and only stores their value if required by the user. This said,  I am providing at each call the values at the previous iteration and retrieve the update values from mfront, which will be used in the newton loop (done externally). As an example, in a incremental small strain formulation, I have the strain incremement based on the actual displacement,  which I use to compute the stress and the tangent stiffness. Similarly, this could be done for internal state variables, at least for what I understood about mfront. 
Thanks again for your support, 
Mauro
